packages/cli/src/util/load-plan.ts

```diff
@@ -220,6 +223,26 @@ const importExpressions = async (
       );
       job.configuration = JSON.parse(configString!);
     }
+    if (stateStr && isPath(stateStr)) {
+      const stateString = await fetchFile(
+        job.id || `${idx}`,
+        rootDir,
+        stateStr,
+        log
+      );
+      job.state = JSON.parse(stateString!);
+    }
+
+    const state = job.state
+    if (typeof state === 'object' && state !== null) {
+      const keys = Object.keys(state);
+      for (const key of keys) {
+        if (typeof state[key] === 'string' && isPath(state[key])) {
+          const fileContent = await fetchFile(job.id || `${idx}`, rootDir, state[key], log);
```

Comment:
@SatyamMattoo What is the use-case for loading nested paths in `state.data`? It seems over-complicated and a bit weird to me

Reply:
Hey @josephjclark, this was for loading file paths available within the state (like `state.data`). The issue asked for loading paths for `data` as well, so if `state.data` or any key of the `state` is a path it will load that file for us.

Reply:
Hmm, I don't think the issue asks for that at all :thinking: 

The issue just asks to load `data` as a single path. That's wrong, as you've pointed out, it's actually the `state` key that needs loading as single path. But it says nothing about recusing through the data to load additional files. That's not a behaviour I think we need.

Reply:
I think the code might be misleading, what it actually does is that if the `state.data` is a file path like `/tmp/data.json` it will load that file for us. This function will load the data (or any key inside the state) for us. 
Are you suggesting we only load paths for the `state` and not for `data` inside the state? If so, I will remove this.

For further clarification this is what the function does:
If any key like data inside the state.json is a file path like:

```
{
    "data": "./data.json"
}
```
And the `data.json`:

{
    "x": 1
}

It will load the content from the `data.json` for us as well.

Reply:
> Are you suggesting we only load paths for the state and not for data inside the state? 

Yes :)

Forget about data - that's a mistake in the original issue. A typo basically. The only key we care about is `state`.

If `step.state` is as string, try to load it as a file and set the resulting JSON to the value of `step.state`.

Reply:
Okay thank you! I have pushed the changes.